boot/zephyr/main.c

```diff
@@ -197,6 +205,31 @@ void main(void)
     }
 #endif
 
+#ifdef CONFIG_BOOT_WAIT_FOR_DFU
```

Comment:
Can this block (CONFIG_BOOT_WAIT_FOR_DFU) be moved to `zephyr_tree/subsys/usb/class/usb_dfu.c` and replaced with a function call like boot_wait_for_usb_dfu. Then we would have a self-contained code (k_poll_event and k_poll_signal stuff) in `usb_dfu.c` and not spread over two repositories.

Reply:
It is good idea to move code within this #ifdef clause to separate function, best to the zephyr. But I think we still want to kept configuration interface here (mean Kconfig).

Reply:
Sounds good. Done.